src/tests/helpers.js

```diff
@@ -1692,6 +1690,126 @@ export const schemaFieldsetScopedCondition = {
   type: 'object',
 };
 
+export const schemaWithConditionalToFieldset = {
+  additionalProperties: false,
+  type: 'object',
+  properties: {
+    work_hours_per_week: {
+      title: 'Hours per week',
+      type: 'string',
+      description: 'Above 30 hours, the Perk>Food options change, and PTO is required.',
+      'x-jsf-presentation': {
+        inputType: 'number',
+      },
+    },
+    pto: {
+      title: 'Time-off (days)',
+      type: 'string',
+      'x-jsf-presentation': {
+        inputType: 'number',
+      },
+    },
+    perks: {
+      additionalProperties: false,
+      properties: {
+        food: {
+          oneOf: [
+            {
+              const: 'lunch',
+              title: 'Lunch',
+            },
+            {
+              const: 'dinner',
+              title: 'Dinner',
+            },
+            {
+              const: 'all',
+              title: 'All',
+              description: 'Every meal',
+            },
+            {
+              const: 'no',
+              title: 'No food',
+            },
+          ],
+          title: 'Food',
+          type: 'string',
+          'x-jsf-presentation': {
+            inputType: 'radio',
+          },
+        },
+        retirement: {
+          oneOf: [
+            {
+              const: 'basic',
+              title: 'Basic',
+            },
+            {
+              const: 'plus',
+              title: 'Plus',
+            },
+          ],
+          title: 'Retirement',
+          type: 'string',
+          'x-jsf-presentation': {
+            inputType: 'radio',
+          },
+        },
+      },
+      required: ['food', 'retirement'],
+      title: 'Perks',
+      type: 'object',
+      'x-jsf-presentation': {
+        inputType: 'fieldset',
+      },
+    },
+  },
+  allOf: [
+    {
+      if: {
+        properties: {
+          work_hours_per_week: {
+            minimum: 30,
+          },
+        },
+        required: ['work_hours_per_week'],
+      },
+      then: {
+        properties: {
+          pto: {
+            $comment: '@BUG: This description does not disappear once activated.',
+            description: 'Above 30 hours, the PTO needs to be at least 20 days.',
+            minimum: 20,
+          },
+          perks: {
+            properties: {
+              food: {
+                description: "Above 30 hours, the 'no' option disappears.",
+                oneOf: [
+                  {
+                    const: 'lunch',
+                    title: 'Lunch',
+                  },
+                  {
+                    const: 'dinner',
+                    title: 'Dinner',
+                  },
+                  {
+                    const: 'all',
+                    title: 'all',
+                  },
+                ],
+              },
+            },
+          },
+        },
+        required: ['pto'],
+      },
+    },
+  ],
+  required: ['perks', 'work_hours_per_week'],
+};
+
```

Comment:
random note, i'm always thankful for https://www.jsonschemavalidator.net/ to let me focus purely on schemas :D

Reply:
TIL about this one. What's the diff between that and https://jsonschemalint.com/?